### PR TITLE
Introduce the first section of the RMC tutorial

### PR DESCRIPTION
**Pull request by tedinski**
### Description of changes: 

Add the first section of the RMC tutorial.

### Resolved issues:

Progress towards #254 

### Call-outs:

* **Test case generation** The source material for CBMC I'm taking inspiration from does "test case generation" but we don't seem to get this output from cbmc viewer, so I've skipped this for now.

### Testing:

* How is this change tested? The new little crate can `cargo test` and `cargo rmc` and `rmc tests/final-form.rs`. This currently isn't in CI. That will be deferred until later.

* Is this a refactor change? no

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [n/a] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: rmc-docs/src/tutorial-first-steps.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+```
+rmc --visualize tests/final-form.rs
+open report/html/index.html
+```
+
+The beginning of the report includes coverage information.
+Clicking through to the file will show fully-covered lines in green.
+Lines not covered by our proof harness will show in red.
+
+1. Try changing the assumption in the proof harness to `x < 2048`. Now the harness won't be testing all posssible cases.
```
Comment (zhassan-aws) [inline rmc-docs/src/tutorial-first-steps.md:184]: An extra 's' in "possible".

File: rmc-docs/src/tutorial-first-steps.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
+# First steps with RMC
+
+> This tutorial expects you to have followed the RMC [installation instructions](./install-guide.md) first.
+
+RMC is unlike the testing tools you may already be familiar with.
+Much of testing is concerned with thinking of new corner cases that need to be covered.
+With RMC, all the corner cases are covered from the start, and the new concern is narrowing down the scope to something managable for the checker.
```
Comment (zhassan-aws) [inline rmc-docs/src/tutorial-first-steps.md:7]: managable -> manageable

File: rmc-docs/src/tutorial/rmc-first-steps/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        #![proptest_config(ProptestConfig::with_cases(10000))]
+        #[test]
+        fn doesnt_crash(x: u32) {
+            estimate_size(x);
+        }
+    }
+// ANCHOR_END: proptest
+}
+
+fn __nondet() -> u32 {
```
Comment (zhassan-aws) [inline rmc-docs/src/tutorial/rmc-first-steps/src/lib.rs:46]: This function might confuse people. Should we include `prelude.rs` instead to "hide" some of the confusion?

File: rmc-docs/src/tutorial-first-steps.md (hunk 1 above)
Comment (tedinski) [reply rmc-docs/src/tutorial-first-steps.md:184]: I really need to get spell check working in vs code :/

File: rmc-docs/src/tutorial/rmc-first-steps/src/lib.rs (hunk 3 above)
Comment (tedinski) [reply rmc-docs/src/tutorial/rmc-first-steps/src/lib.rs:46]: I sort of, kind of, hide it already, at least in that this code doesn't actually appear in the documentation.

But yes, we need to find a good way to address this. Right now it looked like there's no good way to include the prelude! The tests we have, at any rate, have to directly refer to it by relative path.

Fixing this is probably a major developer experience issue. My thought is: fix that, then revise this sample.

I create https://github.com/model-checking/rmc/issues/477 to take this as a note.

Comment (zhassan-aws) [reply rmc-docs/src/tutorial/rmc-first-steps/src/lib.rs:49]: Sounds good.

**Review by zhassan-aws — APPROVED** [review]
Looks great!

**Pull request merged**